test/e2e/network/network_policy.go

```diff
@@ -2004,20 +2202,18 @@ func createNetworkClientPodWithRestartPolicy(f *framework.Framework, namespace *
 			RestartPolicy: restartPolicy,
 			Containers: []v1.Container{
 				{
-					Name:  "client",
-					Image: imageutils.GetE2EImage(imageutils.BusyBox),
+					Name:    "client",
+					Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+					Command: []string{"/bin/sh"},
 					Args: []string{
-						"/bin/sh",
 						"-c",
-						fmt.Sprintf("for i in $(seq 1 5); do nc -vz -w 8 %s.%s %d && exit 0 || sleep 1; done; exit 1",
-							targetService.Name, targetService.Namespace, targetPort),
+						fmt.Sprintf("for i in $(seq 1 5); do /agnhost connect %s.%s:%d --protocol %s --timeout 8s && exit 0 || sleep 1; done; exit 1", targetService.Name, targetService.Namespace, targetPort, connectProtocol),
```

Comment:
Comments that would help future test maintainers: What's the rationale behind the magic numbers here?
- why retry 5 times and not more/less
- why timeout at 8s instead of more/less

Reply:
Those magic numbers were already there, I left them while changing the way we open the socket in a way that supports both TCP and SCTP.

I tried to look back up to https://github.com/kubernetes/kubernetes/commit/3471daf7599604561825c7133aee6cc76408d199 but could not find a proper reasoning behind that.

Reply:
Fair enough, thanks for looking